gdal.cmake

```diff
@@ -301,6 +301,10 @@ if (MSVC)
       CACHE STRING "Postfix to add to the GDAL dll name for debug builds")
   set_target_properties(${GDAL_LIB_TARGET_NAME} PROPERTIES DEBUG_POSTFIX "${GDAL_DEBUG_POSTFIX}")
 endif ()
+if(MINGW)
+    set_target_properties(${GDAL_LIB_TARGET_NAME} PROPERTIES SUFFIX "-${GDAL_SOVERSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
```

Comment:
Overriding `SUFFIX` in this way breaks import lib (`.dll.a`) and  static lib (`.a`).

Reply:
Uhm where do you see this? The suffix isn't added to the import library here from what I can see, and generally I've always seen it added only to the runtime library even in projects which build both static and shared libraries.

Reply:
I've reworked this to only set `RUNTIME_OUTPUT_NAME` which should be safer - I've verified building with both  `BUILD_SHARED_LIBS=ON|OFF` that only the `.dll` gets the suffix this way.

Reply:
Maybe I was wrong about the import lib but not about the static lib.

`RUNTIME_OUTPUT_NAME` is an alternative. However, this will put the version before `CMAKE_<CONFIG>_POSTFIX`...

Reply:
POSTFIXES like `d` etc though are only used in the MSVC world? Alternatively I could go back to setting `SUFFIX`, but conditionally on `BUILD_SHARED_LIBS`, so that static libraries are not affected.

Reply:
> POSTFIXES like `d` etc though are only used in the MSVC world? 

Possibly yes, but it is a user option (passed to CMake at configuration time). And with the Ninja multi-config generator and exported config, there may be valid use cases outside the MSVC world.

This just illustrates the key conflict with re-implementing autotools features in CMake: It often works against CMake features (here: multi-config support, user/toolchain control). Most often I meet this conflict with simultaneous builds of static and shared.

To be clear, I'm not against adding the version suffix. I just want to make sure it doesn't cause trouble for other reasonable (CMake) use cases.

Reply:
Do you see any potential issues with overriding SUFFIX just for BUILD_SHARED_LIBS=TRUE?